### PR TITLE
Changes to parsing behavior, documentation and tests

### PR DESCRIPTION
**Pull request by dtronmans**
## Purpose
- Allow keypoint annotations parsing for COCO using the `use_keypoint_ann` when test annotation json is 
- Revert previous change: previously I allowed all images to contain hard negatives (no ground truth labels) for dense estimation tasks (instance segmentation, detection and keypoints). Now just for coco-2017 keypoints we don't parse these images with no annotations, because over 50% of the images in the dataset don't have keypoints
- Documentation regarding parsing coco-2017 and imagenet-sample to make everything clearer
- Test structure and first few tests for parsing using CLI, we can add more over time depending on edge cases that appear or behavior we want to enforce.
- Added coco keypoints to `test_parse_export_equivalence` and added equivalency testing for semantic segmentation masks (realized I had it for instance segmentation but not semantic segmentation)

### CODE REVIEW

File: luxonis_ml/data/parsers/README.md [hunk 1]
```diff
@@ -62,3 +75,246 @@ This mode:
 - Take 500 random samples from the original train split
 - Take 100 random samples from the original val split
 - Take 0 samples from the test split
+
+## Common datasets for evaluation
```
Comment (klemen1999) [inline luxonis_ml/data/parsers/README.md]: Nitpick: Let's put this whole section one level up so its not nested under LuxonisParser but instead in its own section.

File: luxonis_ml/data/parsers/README.md [hunk 2]
```diff
@@ -62,3 +75,246 @@ This mode:
...
+
+# With a subset of samples
+luxonis_ml data parse ./coco-2017 --split-ratio 1000,200,100
+```
+
+### Corrupted Image Handling
+
+The COCO-2017 train set contains a small number of images known to cause issues. The parser automatically filters these out when parsing the train split. No user action is required.
+
+## Imagenet-sample Dataset Parsing
```
Comment (ptoupas) [inline luxonis_ml/data/parsers/README.md]: Lets don't make this specific to the [ImageNet Sample](https://docs.voxel51.com/dataset_zoo/datasets/imagenet_sample.html#imagenet-sample) dataset.
Lets just talk about the [ImageNet 2012](https://docs.voxel51.com/dataset_zoo/datasets/imagenet_2012.html#imagenet-2012) (or just ImageNet) dataset instead since `ImageNet Sample` is just a subset of that.

**Review by ptoupas — APPROVED** [review]
Left a couple of minor comments, but apart from that it LGTM.
